app/forms/image-upload.tsx

```diff
@@ -335,7 +339,6 @@ export function Component() {
       await deleteDiskCleanup.mutateAsync({ path: { disk: disk.current.id } })
       disk.current = null
     }
-    cleaningUp.current = false
```

Comment:
Because we turned this back to false when cleanup was done, it did not prevent it from running twice in the case where the first attempt finished before the second started.